daemon/cmd/daemon_main.go

```diff
@@ -1132,6 +1136,17 @@ func initEnv(cmd *cobra.Command) {
 		}
 	}
 
+	if option.Config.EnableHostFirewall && len(option.Config.Devices) == 0 {
+		device, err := linuxdatapath.NodeDeviceNameWithDefaultRoute()
+		if err != nil {
+			msg := "Host firewall's external facing device could not be determined. Use --%s to specify."
```

Comment:
On local testing, this `%s` did not format as expected. I think you have to use `Fatalf` below.

Reply:
Ah, I thought Go would complain if `Fatal()` was incorrect and didn't check. I've sent a quick fix.

Thanks a lot for trying it out @gandro!